clc/modules/object-storage-common/src/main/java/com/eucalyptus/objectstorage/exceptions/s3/S3Exceptions.groovy

```diff
@@ -118,6 +123,20 @@ class S3Exception extends ObjectStorageException {
     this.message = description;
     this.status = statusCode;
   }
+
+  def S3Exception(String errorCode, String description, HttpResponseStatus statusCode, String requestMethod) {
+    super();
+    this.requestMethod = requestMethod;
+  }
+  
+  public void setRequestMethod(String requestMethod) {
+    this.requestMethod = requestMethod;
+  }
+
+  public String getRequestMethod() {
+    return requestMethod;
+  }
+
```

Comment:
Getters and setters are optional for groovy class members, they are available automagically.